### PR TITLE
Add ambient pressure

### PR DESCRIPTION
**Pull request by InvincibleRMC**
# Pull Request

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

- Related Issue #
- Closes #

### CODE REVIEW

File: src/float/float_transceiver/float_transceiver.ino [hunk 1]
```diff
@@ -170,9 +170,9 @@ void loop() {
 
     int intComponent = pressure;
     int fracComponent = trunc((pressure - intComponent) * 10000);
-    char judgePacketBuffer[25];
+    char judgePacketBuffer[30];
```
Comment (InvincibleRMC) [inline src/float/float_transceiver/float_transceiver.ino:173]: @benjaminwp18 extract 30 to some constant?

Comment (benjaminwp18) [reply src/float/float_transceiver/float_transceiver.ino]: added

File: src/surface/gui/gui/widgets/float_comm.py [hunk 2]
```diff
@@ -146,3 +161,21 @@ def handle_serial(self, msg: FloatSerial) -> None:
         """
         self.console.moveCursor(QTextCursor.MoveOperation.End)
         self.console.insertPlainText(f'{msg.serial}\n')
+
+    counter = 0
+
+    @pyqtSlot(FloatSingle)
+    def handle_single(self, msg: FloatSingle) -> None:
+        FloatComm.counter += 1
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/float_comm.py]: Make instance variable

File: src/surface/transceiver/transceiver/serial_reader.py [hunk 3]
```diff
@@ -63,14 +77,55 @@ def ros_publish(self, packet: str) -> None:
...
+                self.data_publisher.publish(msg)
         except Exception as e:
             self.get_logger().error(f"Error {e} caught dropping packet")
-            return
-        self.data_publisher.publish(msg)
+
+
+class SerialReaderPacketHandler:
+
+    def __init__(self, queue_size: int = 5) -> None:
```
Comment (benjaminwp18) [inline src/surface/transceiver/transceiver/serial_reader.py:94]:
```suggestion
    def __init__(self, queue_size: int = AVERAGE_QUEUE_LEN) -> None:
```

File: src/surface/transceiver/transceiver/serial_reader.py [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
 
 MILLISECONDS_TO_SECONDS = 1/1000
 SECONDS_TO_MINUTES = 1/60
 MBAR_TO_METER_OF_HEAD = 0.010199773339984
 
+
+AMBIENT_PRESSURE_DEFAULT = 1013.25  # in (mbar)
+
 # AKA the length of the float in (m)
 FLOAT_CONVERSION_FACTOR = 0.635
```
Comment (benjaminwp18) [inline src/surface/transceiver/transceiver/serial_reader.py:22]:
```suggestion

AVERAGE_QUEUE_LEN = 5
```

File: src/surface/transceiver/transceiver/serial_reader.py [hunk 5]
```diff
@@ -63,14 +77,55 @@ def ros_publish(self, packet: str) -> None:
...
+class SerialReaderPacketHandler:
+
+    def __init__(self, queue_size: int = 5) -> None:
+        self.surface_pressure = AMBIENT_PRESSURE_DEFAULT
+        self.surface_pressures: Queue[float] = Queue(queue_size)
 
     @staticmethod
-    def _message_parser(packet: str) -> FloatData:
+    def is_ros_single_message(packet: str) -> bool:
+        ros_single = ROS_PACKET + "SINGLE"
```
Comment (benjaminwp18) [inline src/surface/transceiver/transceiver/serial_reader.py]: Make this a constant

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

**Review by benjaminwp18 — APPROVED** [review]

**Pull request merged**